python/podio/reading.py

```diff
@@ -50,8 +67,12 @@ def get_reader(filename):
     return sio_io.LegacyReader(filename)
 
   if filename.endswith('.root'):
-    if _is_frame_root_file(filename):
+    root_flavor = _determine_root_format(filename)
+    if root_flavor == RootFileFormat.TTREE:
       return root_io.Reader(filename)
-    return root_io.LegacyReader(filename)
+    if root_flavor == RootFileFormat.RNTUPLE:
+      return root_io.RNTupleReader(filename)
+    if root_flavor == RootFileFormat.LEGACY:
+      return root_io.LegacyReader(filename)
```

Comment:
Maybe not needed now because there aren't that many choices but all of this can be changed by a basically one liner instead of all the if - return:
```
return {RootFileFormat.TTREE: root_io.LegacyReader, RootFileFormat.RNTUPLE: root_io.RNTupleReader, RootFileFormat.LEGACY: root_io.LegacyReader}[_determine_root_format(filename)](filename)
```